physics/hierarchical_system_body.hpp

```diff
@@ -58,8 +58,18 @@ HierarchicalSystem<Frame>::ToBarycentric(System& system) {
   auto const semimajor_axis_less_than = [](
       not_null<std::unique_ptr<Subsystem>> const& left,
       not_null<std::unique_ptr<Subsystem>> const& right) -> bool {
-    return left->jacobi_osculating_elements.semimajor_axis <
+    bool const semimajor_axes = left->jacobi_osculating_elements.semimajor_axis &&
            right->jacobi_osculating_elements.semimajor_axis;
+    bool const mean_motions = left->jacobi_osculating_elements.mean_motion &&
+           right->jacobi_osculating_elements.mean_motion;
+    CHECK(semimajor_axes || mean_motions);
```

Comment:
No need for this. Arguably the check for `^` should be in a function in `kepler_orbit.hpp`, called by `Add`.